homeassistant/components/reolink/host.py

```diff
@@ -162,4 +179,140 @@ async def disconnect(self):
 
     async def stop(self, event=None):
         """Disconnect the API."""
+        await self.unregister_webhook()
         await self.disconnect()
+
+    async def subscribe(self) -> bool:
+        """Subscribe to motion events and register the webhook as a callback."""
+        if self.webhook_id is None:
+            if not self.register_webhook():
+                return False
+
+        if self._api.subscribed:
+            _LOGGER.debug(
+                "Host %s: is already subscribed to webhook %s",
+                self._api.host,
+                self._webhook_url,
+            )
+            return True
+
+        if await self._api.subscribe(self._webhook_url):
+            _LOGGER.info(
+                "Host %s: subscribed successfully to webhook %s",
+                self._api.host,
+                self._webhook_url,
+            )
+        else:
+            _LOGGER.error("Host %s: webhook subscription failed", self._api.host)
+            return False
+
+        return True
+
+    async def renew(self) -> bool:
+        """Renew the subscription of the motion events (lease time is set to 15 minutes)."""
+
+        if not self._api.subscribed:
+            _LOGGER.debug(
+                "Host %s: requested to renew a non-existing Reolink subscription, trying to subscribe from scratch",
+                self._api.host,
+            )
+            return await self.subscribe()
+
+        timer = self._api.renewtimer
+        if timer <= 0:
+            _LOGGER.debug(
+                "Host %s: Reolink subscription expired, trying to subscribe again",
+                self._api.host,
+            )
+            return await self._api.subscribe(self._webhook_url)
+        if timer <= SUBSCRIPTION_RENEW_THRESHOLD:
+            if not await self._api.renew():
+                _LOGGER.debug(
+                    "Host %s: error renewing Reolink subscription, trying to subscribe again",
+                    self._api.host,
+                )
+                return await self._api.subscribe(self._webhook_url)
+            _LOGGER.info(
+                "Host %s SUCCESSFULLY renewed Reolink subscription", self._api.host
+            )
+
+        return True
+
+    async def register_webhook(self) -> bool:
+        """Register the webhook for motion events."""
+        self.webhook_id = f"reolink_{self.unique_id.replace(':', '')}"
+        event_id = self.webhook_id
+
+        try:
+            webhook.async_register(
+                self._hass, DOMAIN, event_id, self.webhook_id, self.handle_webhook
+            )
+        except ValueError:
+            _LOGGER.debug(
+                "Error registering webhook %s. Trying to unregister it first and re-register again",
+                self.webhook_id,
+            )
```

Comment:
Why do you have a `ValueError` check here?

Reply:
This came from the custom_component it is based on.
I think a ValueError is generated when the same webhook_id is used twice.
However webhook_ids schould be unique, so the ValueError schould never be raised unless we have a conflict with some custom_integration.

I am fine with removing the except.

Schould I remove it?

Reply:
Yes, please remove this case if we don't expect it to happen. I think this exception is raised if there would be a developer error. In that case it should be fixed earlier in the code.

Reply:
@MartinHjelmare I have removed the except ValueError, thanks for clarifying